src/backend/metal/src/factory.rs

```diff
@@ -483,8 +483,8 @@ impl core::Factory<Resources> for Factory {
     fn create_texture_raw
         (&mut self,
          desc: core::texture::Info,
+         _: core::texture::Mipmap,
```

Comment:
I believe this parameter is meaningless when the `data_opt.is_none()`.
Therefore, we should move it behind the data `Option<>`